source/CommonJobs/CommonJobs.MVC.UI/Scripts/Evaluations/calification.js

```diff
@@ -1,145 +1,265 @@
+                    var cal = calification.Califications[i];
+                    valuesByKey[cal.Key] = cal.Value;
+                }
+            }
+            
+            return valuesByKey;
+        });
+
+        this.groups =_(_(_(data.Template.Items)
+            .groupBy(function (item) {
+                return item.GroupKey;
+            })).map(function (items, key) {
+                var result = {
+                    groupKey: key,
+                    name: groupNames[key],
+                    items: _.map(items, function (item) {
+                        return {
+                            key: item.Key,
+                            text: item.Text,
+                            description: item.Description,
+                            values: _.map(valuesByKeyCollection, function (valuesByKey) {
+                                var valueItem = {
+                                    calificationId: valuesByKey.calificationColumn.calificationId,
+                                    value: ko.observable(valuesByKey[item.Key] || ""),
+                                    editable: self.isValueEditable(valuesByKey),
+                                    showValue: _.find(self.califications, function (calification) {
+                                            return calification.id == valuesByKey.calificationColumn.calificationId;
+                                        }).show
+                                };
+                                self.isDirty.register(valueItem.value);
+                                return valueItem;
+                            })
+                        };
+                    })
+                }
+                result.averages = ko.computed(function () {
+                    var averages = []
+                    for (var i = 0; i < result.items[0].values.length; i++) {
+                        for (var key in result.items) {
+                            if (!averages[i]) {
+                                averages[i] = {
+                                    count: 0,
+                                    total: 0,
+                                    show: self.califications[i].show
+                                };
+                            }
+                            value = parseFloat(result.items[key].values[i].value());
+                            if (value) {
+                                averages[i].count++;
+                                averages[i].total += value;
+                            }
+                        }
+                    }
+                    return _.map(averages, function (column) {
+                        return {
+                            value: (column.total) ? (column.total / column.count).toFixed(1) : 0,
+                            show: column.show
+                        };
+                    });
+                });
+                return result;
+            }))
+            .value();
```

Comment:
This expression looks way too complex -- what is it that we're trying to achieve here?